src/libfetchers/mercurial.cc

```diff
@@ -120,7 +120,7 @@ struct MercurialInputScheme : InputScheme
     {
         auto url = parseURL(getStrAttr(input.attrs, "url"));
         if (url.scheme == "file" && !input.getRef() && !input.getRev())
-            return url.path;
+            return url.renderPath(/*encode=*/false);
```

Comment:
`renderUrlPathEnsureNoSlash` since it is going to be file path? your call though